api/server.py

```diff
@@ -19,12 +21,36 @@
 app.config["SECRET_KEY"] = "37Y,=i9.,U3RxTx92@9j9Z[}"
 Session(app)
 
-CORS(app, resources={r"/*": {"origins": "http://localhost:{}".format(REACT_PORT)}})
-
+CORS(app, supports_credentials=True, resources={r"/*": {"origins": "http://localhost:{}".format(REACT_PORT)}})
 
 def run(port, hostname, debug_mode):
-    app.run(host=hostname, port=port, debug=debug_mode)
-
+    app.run(host=hostname, port=port, debug=debug_mode, threaded=True)
+
+@app.route('/login', methods=["POST", "GET"])
+def login():
+    if request.method == 'POST':
+        data = request.get_json()
+        print(data)
+        session['username'] = data['username']
+        session['dataset'] = data['dataset']
+        session['purpose'] = data['purpose']
+        # TODO use key from dataset to select data
+        meta_path_loader = MetaPathLoaderDispatcher().get_loader("Rotten Tomato")
+        meta_paths = meta_path_loader.load_meta_paths()
+        session['meta_path_distributor'] = RandomMetaPathSelector(meta_paths=meta_paths)
+        session['meta_path_id'] = 1
+        session['rated_meta_paths'] = []
+    return jsonify({'status': 200})
+
+
+@app.route('/logout')
+def logout():
+    # TODO better filenaming
+    filename = '{}_{}_{}.json'.format(session['dataset'],session['purpose'],session['username'])
```

Comment:
Add some unique ID. Otherwise the files with the same purpose and username get overwritten.